src/puppeteer_utils.js

```diff
@@ -143,32 +164,75 @@ const enqueueRequestsFromClickableElements = async (page, selector, purls, reque
  * @return {Promise<RequestOperationInfo[]>}
  *   Promise that resolves to an array of {@link RequestOperationInfo} objects.
  * @memberOf puppeteer
  */
-const enqueueLinks = async (page, selector, pseudoUrls, requestQueue) => {
+const enqueueLinks = async (page, selector, requestQueue, pseudoUrls = []) => {
+    // TODO: Remove after v1.0.0 gets released.
+    // Check for pseudoUrls as a third parameter.
+    if (Array.isArray(requestQueue)) {
+        if (logDeprecationWarning) {
+            log.warning('Argument "pseudoUrls" as the third parameter to enqueueLinks() is deprecated. '
+                + 'Use enqueueLinks(page, selector, requestQueue, pseudoUrls) instead. "pseudoUrls" are now optional.');
+            logDeprecationWarning = false;
+        }
+        const tmp = requestQueue;
+        requestQueue = pseudoUrls;
+        pseudoUrls = tmp;
+    }
+
     checkParamOrThrow(page, 'page', 'Object');
     checkParamOrThrow(selector, 'selector', 'String');
-    checkParamOrThrow(pseudoUrls, 'purls', 'Array');
     checkParamPrototypeOrThrow(requestQueue, 'requestQueue', [RequestQueue, RequestQueueLocal], 'Apify.RequestQueue');
+    checkParamOrThrow(pseudoUrls, 'pseudoUrls', 'Array');
+
+    // Construct pseudoUrls from input where necessary.
+    const pseudoUrlInstances = pseudoUrls.map((item, idx) => {
+        // Get pseudoUrl instance from cache.
+        let pUrl = enqueueLinksCache.get(item);
+        if (pUrl) return pUrl;
+        // Nothing in cache, make a new instance.
+        checkParamOrThrow(item, `pseudoUrls[${idx}]`, 'Object|String');
+        if (item instanceof PseudoUrl) pUrl = item;
+        else if (typeof item === 'string') pUrl = new PseudoUrl(item);
+        else pUrl = new PseudoUrl(item.purl, _.omit(item, 'purl'));
+        // Manage cache
+        enqueueLinksCache.set(item, pUrl);
+        if (enqueueLinksCache.size > MAX_ENQUEUE_LINKS_CACHE_SIZE) {
+            const key = enqueueLinksCache.keys().next().value;
+            enqueueLinksCache.delete(key);
+        }
+        return pUrl;
+    });
 
     /* istanbul ignore next */
     const getHrefs = linkEls => linkEls.map(link => link.href).filter(href => !!href);
     const urls = await page.$$eval(selector, getHrefs);
-    const requests = [];
+    let requests = [];
 
-    urls.forEach((url) => {
-        pseudoUrls
-            .filter(purl => purl.matches(url))
-            .forEach(purl => requests.push(purl.createRequest(url)));
-    });
+    if (pseudoUrlInstances.length) {
+        urls.forEach((url) => {
+            pseudoUrlInstances
+                .filter(purl => purl.matches(url))
+                .forEach(purl => requests.push(purl.createRequest(url)));
+        });
+    } else {
+        requests = urls.map(url => ({ url }));
+    }
 
-    return Promise.mapSeries(requests, request => requestQueue.addRequest(request));
+    const requestOperationInfos = [];
```

Comment:
It looked so much better with mapSeries :D

Reply:
Yes it did! I'm just trying to get rid of Bluebird everywhere.